source/Playnite/ViewModels/MainViewModelBase.cs

```diff
@@ -586,28 +586,28 @@ public async Task UpdateLibrary(LibraryPlugin plugin)
                 var addedGames = ImportLibraryGames(plugin, token);
                 if (AppSettings.ScanLibInstallSizeOnLibUpdate)
                 {
-                    UpdateLibraryInstallSizes(token);
+                    UpdateGamesInstallSizes(token, Database.Games, LOC.ProgressScanningGamesInstallSize);
                 }
 
                 return addedGames;
             }, AppSettings.DownloadMetadataOnImport);
         }
 
-        public void UpdateLibraryInstallSizes(CancellationToken token)
+        public void UpdateGamesInstallSizes(CancellationToken token, IEnumerable<Game> games, string progressMessageLocKey)

```

Comment:
Please add token cancellation check at the beginning of this method since it could be in cancelled state from when cancellation was initiated during game import process.

Reply:
Done in https://github.com/JosefNemec/Playnite/pull/3100/commits/d224016ca7745e02b9ecf2fe2e645932deeb4118